src/diff/index.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Command, flags } from "@oclif/command";
+
+import { findApiChanges, diffRaml } from "./diffProcessor";
+import { NodeDiff } from "./jsonDiff";
+
+export { findApiChanges, diffRaml, RamlDiff } from "./diffProcessor";
+export { NodeDiff } from "./jsonDiff";
+export default class DiffCommand extends Command {
+  static description = `Takes two API spec files as input and outputs the differences.
+By default, a ruleset is applied to determine if changes are breaking. Exit status is:
+  0 - all changes are non-breaking
+  1 - any changes are breaking
+  2 - evaluation could not be completed
+
+The ruleset flag is used to evaluate a custom ruleset in place of the default rules. The diff-only flag disables evaluation against any ruleset.
+`;
+
+  static flags = {
+    // Add --version flag to show CLI version
+    version: flags.version({ char: "v" }),
+    // Add --help flag to show CLI version
+    help: flags.help({ char: "h" }),
+    ruleset: flags.string({
+      char: "r",
+      description: "Path to ruleset to apply to diff",
+      env: "DIFF_RULESET",
+      exclusive: ["diff-only"]
+    }),
+    "diff-only": flags.boolean({
+      description:
+        "Only show differences without evaluating a ruleset. The exit status in this mode is 0 for no changes, 1 for any difference and 2 when unsuccessful.",
+      default: false
+    })
+  };
+
+  static args = [
+    {
+      name: "apiSpecBasePath",
+      required: true,
+      description: "The base API spec file for the comparison"
+    },
+    {
+      name: "apiSpecNewPath",
+      required: true,
+      description:
+        "The new version of the API spec for comparison against the base version"
+    }
+  ];
+
+  async run(): Promise<void> {
```

Comment:
Not a huge fan of the flow of this.  Kind of flows like a switch statement with ifs, where they are mutually exclusive by simple virtue of exits at the end of the if.  Would it make more sense to convert to a switch statment here and hole exit codes until the end?

Reply:
I don't love it either, but I'm staring at it and struggling to find something I like better. I can't see where a switch would come in...there's no place here where there's three branches based on a single value. There's one behavior for diff-only and a different one without it. I can break it up, but it's a single screen of code so it just feels like more jumping around to follow it. Open to suggestions, I'm just not seeing something that reads easier to me.

Reply:
Still don't love the flow here and think we should be thinking about what we want to do for these kinds of main flow logic so it flows a little more smoothly.